<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>931</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="image_label">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>80</y>
      <width>720</width>
      <height>480</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QSlider" name="horizontalSlider">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>80</y>
      <width>160</width>
      <height>22</height>
     </rect>
    </property>
    <property name="maximum">
     <number>255</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>120</y>
      <width>111</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="img_inf">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>20</y>
      <width>341</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>50</y>
      <width>71</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>二值化閥值</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>170</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>水平位移量</string>
    </property>
   </widget>
   <widget class="QSlider" name="horizontal_displacement">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>200</y>
      <width>160</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-800</number>
    </property>
    <property name="maximum">
     <number>800</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>280</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>垂直位移量</string>
    </property>
   </widget>
   <widget class="QSlider" name="Vertical_displacement">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>310</y>
      <width>160</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-800</number>
    </property>
    <property name="maximum">
     <number>800</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="horizontal_label">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>230</y>
      <width>61</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="Vertical_label">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>350</y>
      <width>61</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>400</y>
      <width>71</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>旋轉角度</string>
    </property>
   </widget>
   <widget class="QSlider" name="rotate_horizontalSlider">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>440</y>
      <width>160</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-360</number>
    </property>
    <property name="maximum">
     <number>360</number>
    </property>
    <property name="sliderPosition">
     <number>0</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="rotate_label">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>470</y>
      <width>51</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>500</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>偏移倍率</string>
    </property>
   </widget>
   <widget class="QSlider" name="scale_horizontalSlider">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>540</y>
      <width>160</width>
      <height>22</height>
     </rect>
    </property>
    <property name="maximum">
     <number>20</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="scale_label">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>580</y>
      <width>47</width>
      <height>12</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="cornerharris_label">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>620</y>
      <width>71</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>角點檢測</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="cornerharris_lineEdit">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>660</y>
      <width>113</width>
      <height>22</height>
     </rect>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1200</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="file">
    <property name="title">
     <string>檔案</string>
    </property>
    <addaction name="read_img"/>
    <addaction name="save_img"/>
   </widget>
   <widget class="QMenu" name="setting">
    <property name="title">
     <string>設定</string>
    </property>
    <addaction name="ROI"/>
    <addaction name="Histogram"/>
    <addaction name="Color_space"/>
   </widget>
   <widget class="QMenu" name="image_Preprocessing">
    <property name="title">
     <string>影像處理</string>
    </property>
    <addaction name="Thresholding"/>
    <addaction name="Histogram_equalization"/>
    <addaction name="GaussianBlur_3"/>
    <addaction name="Canny"/>
    <addaction name="medianblur"/>
    <addaction name="displacement"/>
    <addaction name="rotate"/>
    <addaction name="affine_transform"/>
    <addaction name="perspective_transform"/>
    <addaction name="cornerharris"/>
    <addaction name="contour_action"/>
   </widget>
   <addaction name="file"/>
   <addaction name="setting"/>
   <addaction name="image_Preprocessing"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="read_img">
   <property name="text">
    <string>寫入影像</string>
   </property>
  </action>
  <action name="ROI">
   <property name="text">
    <string>設定ROI</string>
   </property>
  </action>
  <action name="Histogram">
   <property name="text">
    <string>顯示影像直方圖</string>
   </property>
  </action>
  <action name="Color_space">
   <property name="text">
    <string>顯示或改變色彩空間</string>
   </property>
  </action>
  <action name="Thresholding">
   <property name="text">
    <string>影像二值化</string>
   </property>
  </action>
  <action name="Histogram_equalization">
   <property name="text">
    <string>直方圖等化</string>
   </property>
  </action>
  <action name="save_img">
   <property name="text">
    <string>儲存圖片</string>
   </property>
  </action>
  <action name="actiondada">
   <property name="text">
    <string>dada</string>
   </property>
  </action>
  <action name="GaussianBlur_2">
   <property name="text">
    <string>453453</string>
   </property>
  </action>
  <action name="GaussianBlur_3">
   <property name="text">
    <string>低通濾波</string>
   </property>
  </action>
  <action name="Canny">
   <property name="text">
    <string>高通濾波</string>
   </property>
  </action>
  <action name="medianblur">
   <property name="text">
    <string>去除雜訊</string>
   </property>
  </action>
  <action name="displacement">
   <property name="text">
    <string>位移</string>
   </property>
  </action>
  <action name="rotate">
   <property name="text">
    <string>旋轉</string>
   </property>
  </action>
  <action name="affine_transform">
   <property name="text">
    <string>仿射轉換</string>
   </property>
  </action>
  <action name="perspective_transform">
   <property name="text">
    <string>透視投影轉換</string>
   </property>
  </action>
  <action name="cornerharris">
   <property name="text">
    <string> 角點檢測</string>
   </property>
  </action>
  <action name="contour_action">
   <property name="text">
    <string>輪廓檢測</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
